server/decorators/expandedDeliusServiceUserDecorator.test.ts

```diff
@@ -0,0 +1,293 @@
+              district: 'City of London',
+              county: 'Greater London',
+              from: '2021-01-01',
+              to: null,
+              noFixedAbode: false,
+            }
+            const addressWithNullFromValue = {
+              addressNumber: 'Flat 10',
+              buildingName: null,
+              streetName: 'Test Walk',
+              postcode: 'SW16 1AQ',
+              town: 'London',
+              district: 'City of London',
+              county: 'Greater London',
+              from: null,
+              to: null,
+              noFixedAbode: false,
+            }
+
+            const serviceUser = new ExpandedDeliusServiceUserDecorator(
+              expandedDeliusServiceUserFactory.build({
+                contactDetails: {
+                  addresses: [oldAddress, newAddress, addressWithNullFromValue],
+                },
+              })
+            )
+
+            expect(serviceUser.address).toEqual([
+              'Flat 10 Test Walk',
+              'London',
+              'City of London',
+              'Greater London',
+              'SW16 1AQ',
+            ])
+          })
+        })
+
+        describe('when all addresses have a null "from" field', () => {
+          it('returns "Not found" as we are unable to determine the most recent', () => {
+            const firstAddressNoFromDate = {
+              addressNumber: 'Flat 10',
+              buildingName: null,
+              streetName: 'Test Walk',
+              postcode: 'SW16 1AQ',
+              town: 'London',
+              district: 'City of London',
+              county: 'Greater London',
+              from: null,
+              to: null,
+              noFixedAbode: false,
+            }
+            const secondAddressNoFromDate = {
+              addressNumber: 'Flat 10',
+              buildingName: null,
+              streetName: 'Test Walk',
+              postcode: 'SW16 1AQ',
+              town: 'London',
+              district: 'City of London',
+              county: 'Greater London',
+              from: null,
+              to: null,
+              noFixedAbode: false,
+            }
+
+            const serviceUser = new ExpandedDeliusServiceUserDecorator(
+              expandedDeliusServiceUserFactory.build({
+                contactDetails: {
+                  addresses: [firstAddressNoFromDate, secondAddressNoFromDate],
+                },
+              })
+            )
+
+            expect(serviceUser.address).toEqual(['Not found'])
+          })
+        })
+      })
+
+      describe('when there is no address returned from nDelius', () => {
+        it('returns "Not found"', () => {
+          const serviceUser = new ExpandedDeliusServiceUserDecorator(
+            expandedDeliusServiceUserFactory.build({
+              contactDetails: {
+                addresses: [],
+              },
+            })
+          )
+
+          expect(serviceUser.address).toEqual(['Not found'])
```

Comment:
I reckon that deciding to display a "not found" string is beyond the responsiblities of a model. How about returning `null` from this method in the case where there's no address, and then the presenter can decide what to do with that?

Reply:
Fair point - have changed this and started handling in the presenter.